internal/http/check.go

```diff
@@ -0,0 +1,210 @@
+		logger.WithError(err).Error("error marshalling the check state")
+		http.Error(w, "error marshalling the check state", http.StatusInternalServerError)
+		return
+	}
+	w.Write(out)
+}
+
+// RunAndServe implements the behavior needed by the sdk for a check runner to
+// execute a check.
+func (c *Check) RunAndServe() {
+	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		if c.shuttedDown == nil {
+			w.Header().Set("Content-Type", "application/json")
+			w.Write([]byte(`"OK"`))
+		} else {
+			// the server is shutting down
+			http.Error(w, "shuttingDown", http.StatusServiceUnavailable)
+		}
+	})
+	http.HandleFunc("/run", c.ServeHTTP)
+	c.Logger.Info(fmt.Sprintf("Listening at %s", c.server.Addr))
+	go func() {
+		if err := c.server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+			c.Logger.WithError(err).Error("Starting http server")
+		}
+		c.Logger.Info("Stopped serving new connections.")
+		if c.shuttedDown != nil {
+			c.Logger.Info("Notifying shuttedDown")
+			c.shuttedDown <- 1
+			c.Logger.Info("Notified shuttedDown")
+		}
+	}()
+
+	s := <-c.exitSignal
+
+	c.Logger.WithField("signal", s.String()).Info("Stopping server")
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second) // TODO: Allow configure value.
+	defer cancel()
+	if err := c.server.Shutdown(ctx); err != nil {
+		c.Logger.WithError(err).Error("Shutting down server")
+	}
+	c.Logger.Info("Finished RunAndServe.")
+}
+
+type Job struct {
+	CheckID      string            `json:"check_id"`      // Required
+	StartTime    time.Time         `json:"start_time"`    // Required
+	Image        string            `json:"image"`         // Required
+	Target       string            `json:"target"`        // Required
+	Timeout      int               `json:"timeout"`       // Required
+	AssetType    string            `json:"assettype"`     // Optional
+	Options      string            `json:"options"`       // Optional
+	RequiredVars []string          `json:"required_vars"` // Optional
+	Metadata     map[string]string `json:"metadata"`      // Optional
+	RunTime      int64
+}
+
+// Shutdown is needed to fulfil the check interface and in this case we are
+// shutting down the http server and waiting
+func (c *Check) Shutdown() error {
+	// Send the exit signal to shutdown the server.
+	c.exitSignal <- syscall.SIGTERM
+
+	c.shuttedDown = make(chan int)
+	// Wait for the server to shutdown.
+	c.Logger.Info("Shutdown: waiting for shuttedDown")
+	<-c.shuttedDown
+	c.Logger.Info("Shutdown:shutted down")
+	return nil
+}
+
+// NewCheck creates new check to be run from the command line without having an agent.
+func NewCheck(name string, checker Checker, logger *log.Entry, conf *config.Config) *Check {
+	c := &Check{
+		Name:        name,
+		Logger:      logger,
+		config:      conf,
+		exitSignal:  make(chan os.Signal, 1),
+		port:        *conf.Port,
+		shuttedDown: nil,
+	}
+	c.server = &http.Server{Addr: fmt.Sprintf(":%d", c.port)}
```

Comment:
The server is only used in RunAndServe. I don't think it needs to belong to the `Check` struct.

Reply:
done 877c330718594bef1468b09846b8dbdfe4e3f388